app2.js

```diff
@@ -60,79 +55,99 @@ newGameBtn.addEventListener("click", () => {
     console.log(currentDice);
-    game(currentDice);
+    
+    if (currentDice !== 1 && hold1Score === 0) {    
+        if (currentPlayer == 1) {
+            player1Score = player1Score + currentDice;
+            currentScore1.textContent = player1Score;
+        } else {
+            player2Score = player2Score + currentDice;
+            currentScore2.textContent = player2Score;
+        }
+    } else if (currentDice !== 1 && hold1Score !== 0) {    
+        if (currentPlayer == 1) {
+            let afterHoldScore1 = 0;
+            player1Score = hold1Score + currentDice ;
+            currentScore1.textContent = player1Score;
+        } else {
+            player2Score = hold2Score + currentDice;
+            currentScore2.textContent = player2Score;
+        }
+    } else if (currentDice === 1) {
+        currentScore1.textContent = (currentPlayer === 1) ? "You Lost" : "You Win!";
+        currentScore2.textContent = (currentPlayer === 2) ? "You Lost" : "You Win!";
+    }
+    doesWin();
 });
 
-const holdScore1 = document.querySelector("#holdScore1");
-const holdScore2 = document.querySelector("#holdScore2");
+holdBtn.addEventListener("click", () => {
+    holdFunc();
+});
 
-const holdFunc = () => {
-    //ternary operator if/else as below
-    currentPlayer = (currentPlayer === 1) ? 2 : 1;
-    // everything on the left of there is the test
-    // left is the true branch
-    // right side is after colon is False branch
-    switchBg();
 
+//holdScore kendisine eklenmesi gerekiyor
+
+const holdFunc = () => {
     if(currentPlayer === 1){
+        hold1Score = player1Score;
         holdScore1.textContent = player1Score;
         currentScore1.textContent = 0;
-   
+        p1Bg.style.backgroundColor = "transparent";
+        p2Bg.style.backgroundColor = "white";
     } else {
+        hold2Score = player2Score;
         holdScore2.textContent = player2Score;
         currentScore2.textContent = 0;
+        p1Bg.style.backgroundColor = "transparent";
+        p2Bg.style.backgroundColor = "white";
     }
-
-    console.log(currentPlayer);
-    console.log(player1Score);
-    console.log(player2Score);
-    console.log("holdScore");
+    currentPlayer = (currentPlayer === 1) ? 2 : 1;
+    switchBg();
 }
 
-const holdBtn = document.querySelector("#holdBtn");
-
-holdBtn.addEventListener("click", () => {
-    holdFunc();
-    switchBg();
-});
+const doesWin = () => {
+    if (player1Score >= 20) {
+        currentScore1.textContent = "You Win!";
+        currentScore2.textContent = "You Lost!";
+    } else if (player2Score >= 20) {
+        currentScore1.textContent = "You Lost!";
+        currentScore2.textContent = "You Win!";    
+    }
+}
 
+const  switchBg = () => { 
+    if (currentPlayer === 1) {
+        p1Bg.style.backgroundColor = "white";
+        p2Bg.style.backgroundColor = "transparent";
+    } else {
+        p1Bg.style.backgroundColor = "transparent";
+        p2Bg.style.backgroundColor = "white";
+    }
+}
```

Comment:
In fact doesn't this do what 103, 104, 109 and 110 do? Looks like you can save a few lines of code?

Reply:
yeap i saved 4 lines :)